addons/html_editor/static/src/main/chatgpt/chatgpt_plugin.js

```diff
@@ -128,7 +124,9 @@ export class ChatGPTPlugin extends Plugin {
             // If the dialog is opened on a small screen, remove all selection
             // because the selection can be seen through the dialog on some devices.
             // TODO: Find a better way and avoid modifying range
-            this.document.getSelection()?.removeAllRanges();
+            // when it's a dropdown that opens the dialog, you have to wait for the next microtick
```

Comment:
1)
comment:
```
HACK: In the case of opening through dropdown:
- when dropdown open, it keep the element focused before the open
- when opening the dialog through the dropdown, the dropdown closes
- upon close, the generic code of the dropdown sets focus on the kept element (in our case, the editable)
- we need to remove the range after the generic code of the dropdown is triggered so we hack it by removing the range in the next tick
```
2)
I would move the comment related to removeAllRange inside the `then`